pkg/machine/vmconfigs/machine.go

```diff
@@ -181,11 +181,14 @@ func (mc *MachineConfig) Remove(machines map[string]bool, saveIgnition, saveImag
 
 	rmFiles := []string{
 		mc.configPath.GetPath(),
-		readySocket.GetPath(),
 		gvProxySocket.GetPath(),
 		apiSocket.GetPath(),
 		logPath.GetPath(),
 	}
+	if mc.Capabilities.HasReadyUnit {
```

Comment:
This caused a failure to me bc I already had a machine on my laptop. When trying to delete it to init another one, it failed bc mc.Capabilities is nil

Reply:
> My concern is if one day podman will use our stuff, everything will break up for them bc there is no Capabilities data, and the user is forced to recreate the podman machines. Atleast if we put a guard we are safe in any case.

Ah good point!

> I'm ok with the GetHasReadyUnit()... maybe the if part should return true? If there is no capabilities (if caps == nil {) specified i assume it is an old (podman??) machine and should use the old code. WDYT?

Yes, this was my intent but I messed that up :) I’ll fix it.